### PR TITLE
 update(component): carousel redesign

### PR DESCRIPTION
**Pull request by gepifanio**

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit 508de38855e06d7bb5ffdb854e165c201a589aaf

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/5d2d36d62b31ec00087df951

File: components/carousel/index.css [hunk 1]
```diff
@@ -88,12 +91,23 @@
...
+      margin-bottom: .5rem;
+    }
   }
 
   &__description {
-    color: #000;
+    color: #fff;
     font-size: 1rem;
-    line-height: 1.5;
+    line-height: 1.375rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: use unitless value.

File: components/carousel/index.css [hunk 2]
```diff
@@ -146,27 +159,30 @@
...
+      padding: .42rem 1rem .4rem 1rem;
       cursor: pointer;
       display: block;
       text-decoration: none;
-      color: rgb(var(--col-background-primary-navy));
+      color: rgb(var(--col-background-primary-white));
       font-weight: var(--fw-bold);
-      font-size: 1.15rem;
+      font-size: 1.2rem;
+      line-height: 1.5625rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: use unitless value.

File: components/carousel/index.css [hunk 3]
```diff
@@ -76,10 +75,14 @@
...
     margin-bottom: 1rem;
-    font-size: 1.5rem;
-    color: rgb(var(--col-navy-dark));
-    font-weight: var(--fw-bold);
-    line-height: 1;
+
+    a {
+      font-family: var(--ff-normal);
+      font-size: 2.075rem;
+      line-height: 2.8031rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: use unitless value.

File: components/carousel/index.css [hunk 4]
```diff
@@ -175,21 +191,20 @@
...
-      flex-basis: 30%;
+      flex-basis: 40%;
       margin-left: 1rem;
       flex-direction: column-reverse;
       justify-content: flex-end;
     }
   }
 
   &__cta {
+    margin-top: 1.375rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: Using margin-top can cause alignment issues when certain elements/components are shown/hidden.  Using margin-bottom allows all components to always align to the top.

Remove this and add `margin-bottom: 2.8rem;` to `carousel__description`

File: components/carousel/index.css [hunk 5]
```diff
@@ -76,10 +75,14 @@
 
   &__title {
     margin-bottom: 1rem;
-    font-size: 1.5rem;
-    color: rgb(var(--col-navy-dark));
-    font-weight: var(--fw-bold);
-    line-height: 1;
+
+    a {
```
Comment (TimMansell) [inline components/carousel/index.css]: this can be moved to `carousel__title-link`

File: components/carousel/index.css (hunk 4 above)
Comment (TimMansell) [reply components/carousel/index.css]: We don't need any of the `&__cta` classes as we will address button changes in #1205

File: components/carousel/index.css [hunk 6]
```diff
@@ -146,27 +159,30 @@
...
-        background: rgb(var(--col-background-primary-navy));
-        color: rgb(var(--col-background-primary-white)) !important;
+        background: rgb(var(--col-background-primary-white));
+        color: rgb(var(--col-background-primary-navy));
       }
     }
 
     @media (--bp-desktop) {
       display: block;
+      margin-top: 1.5rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: remove this and update `carousel__controls` margin-bottom to 2.5rem;

File: components/carousel/index.css [hunk 7]
```diff
@@ -58,11 +51,17 @@
         flex-direction: row-reverse;
         display: flex;
       }
+
+      @media (--bp-x-desktop) {
+        padding: 1.5rem 7.125rem 1.5rem 7.125rem;
```
Comment (TimMansell) [inline components/carousel/index.css]: if top/bottom & left/right are the same you can use the shorthand values.

```
padding: 1.5rem 7.125rem;
```

File: components/carousel/index.css [hunk 8]
```diff
@@ -58,11 +51,17 @@
...
+
+      @media (--bp-wide) {
+        padding: 1.5rem 14.25rem 1.5rem 14.25rem;
+      }
     }
 
     @media (--bp-desktop) {
-      width: 75%;
-      position: absolute;
       z-index: 2;
```
Comment (TimMansell) [inline components/carousel/index.css]: use z-index variable

**Comment by TimMansell** [comment]
IE11 

Screen Shot 2019-07-16 at 2 48 24 pm

**Comment by TimMansell** [comment]
The CSS implementation for adding the ellipsis only works in Chrome so I removed it and implemented a solution in the vue truncate filter.

@jonathan-doc will get back to you on the character limits to use.

**Comment by jaredh-um** [comment]
@gepifanio The image is no longer changing for me, so I can't really review for now

**Comment by TimMansell** [comment]
@jaredh-um I've fixed the issue with the images not changing.

**Comment by jaredh-um** [comment]
@TimMansell There's some weird interactions with the progress/pausing behaviour. 

- Pauses like it should when you click a story link or use the different controls - but then restarts when you move focus out of the browser window.

- Clicking the image while playing drops the progress back to something like 5%

- Clicking the image when it's paused drops it back to 0%

- The image should be non-interactive

- Allowing the progress bar to naturally move on to the next story, the progress bar only drops back to about 5-10% before it resumes incrementing

- Seems to get stuck at 100% if it is playing and you switch to another tab (pause button visible so it should be playing)

- Hit areas for controls could be bigger

Otherwise it's looking pretty good!

**Comment by jonathan-doc** [comment]
@gepifanio @TimMansell 
Some minor changes:
- [ ] Set margin-bottom of **carousel__controls** to **1rem** (it already seems like it is for the Desktop/Full variant, but the Mobile/Compact margin is way too tall)
- [ ] **Remove margin-bottom** for **carousel__navigation**, since **carousel__panel-container** now has its 1.5rem padding
- [ ] Set **carousel__panel-container**'s padding-bottom to **2rem** (also change it for the low breakpoint, where it's set to 1.5 rem (the other padding values should remain unchanged)

**Comment by TimMansell** [comment]
Happy for this to be merged if @jaredh-um & @jonathan-doc are fine with it too.

**Pull request merged**